File: 05-cifrado-cesar/main.js
const encriptar = () => {
    let abc = "abcdefghijklmnñopqrstuvwxyz";
    let mensaje = prompt("Ingresa el mensaje que quieres encriptar");
    let desplazamiento = Number(prompt("Ingresa el desplazamiento requerido"));
    let resultado = "";

    for (let i = 0; i < mensaje.length; i++) {
        let posActual = abc.indexOf(mensaje[i]);
        let posDeseada = posActual + desplazamiento
        let posNueva = posDeseada < abc.length ? posDeseada : posDeseada % abc.length;
        let letra = abc[posNueva];
        
        resultado += letra;
    }

    return resultado;
}

console.log(encriptar());

File: 05-cifrado-cesar/main.py
def cifrado(mensaje, desplazamiento, abecedario):
	mensaje_cifrado = ""
	
    for letra in mensaje:
	    pos = abc.find(letra)
	    final = pos + desplazamiento
	    conversion = abc[final]
	    mensaje_cifrado += conversion
    
    return mensaje_cifrado


def descifrar(mensaje, desplazamiento, abecedario):
	mensaje_descifrado = ""
	
    for letra in mensaje:
		pos = abc.find(letra)
		final = pos - desplazamiento
		conversion = abc[final]
		mensaje_descifrado += conversion

    return mensaje_descifrado


abc= "abcdefghijklmnñopqrstuvwxyzabcdefghijklmnñopqrstuvwxyz"
print("Bienvenido al software para encriptar mensajes")

mensaje = input("escribe el mensaje que quieres encriptar")
desplazamiento =int( input("Escribe la cantidad de desplazamiento 1-27"))

c = cifrado(mensaje,desplazamiento,abc)
print("tu mensaje cifrado es: ",c)

d = descifrar(c,desplazamiento,abc)
print("tu mensaje descifrado es: ",d)